naxsi_src/naxsi_runtime.c

```diff
@@ -181,6 +181,73 @@ ngx_http_naxsi_rawbody_parse(ngx_http_request_ctx_t* ctx,
 unsigned char*
 ngx_utf8_check(ngx_str_t* str);
 
+/*
+ * variables to use pcre2
+ */
+static pcre2_match_data       *ngx_pcre2_match_data;
+static ngx_uint_t              ngx_pcre2_match_data_size;
+
+/*
+ * helper function to use pcre2
+ */
+ngx_int_t
+ngx_pcre2_exec(ngx_regex_t *re, unsigned char* str, unsigned int len, ngx_int_t tmp_idx, int *captures, ngx_uint_t size)
+{
```

Comment:
this function should be guarded by `#if defined nginx_version && (nginx_version >= 1021005)` like below.

Reply:
Hi,
I initially thought to use a `#if (NGX_PCRE2)` like in https://github.com/nginx/nginx/blob/master/src/core/ngx_regex.c#L79, but I was not sure if NGX_PCRE2 was available at that point, so i left that function 'unguarded'.
So yes, I agree with your suggestion: should I correct it directly? (Sorry, I'm really new on github!)

Reply:
just add an extra commit for this.